src/brokers/broker-details/components/Overview/Metrics/components/CardBrokerCPUUsageMetrics/CardBrokerCPUUsageMetrics.container.tsx

```diff
@@ -70,18 +62,22 @@ export const CardBrokerCPUUsageMetricsContainer: FC<
 
   // If we have both `timespan` and `defaultTimespan`, `timespan` takes precedence
   // Limit the number of samples so that the step size doesn't fall below minStep
-  const maxSamplesForSpan = defaultSamples || getMaxSamplesForSpan(span);
+  const maxSamplesForSpan =
+    defaultSamples || getMaxSamplesForSpan(parsePrometheusDuration(state.span));
   const [samples, setSamples] = useState(maxSamplesForSpan);
 
   // Define this once for all queries so that they have exactly the same time range and X values
   const endTime = xDomain?.[1];
 
   // If provided, `timespan` overrides any existing span setting
   useEffect(() => {
-    if (span) {
-      setSamples(defaultSamples || getMaxSamplesForSpan(span));
+    if (state.span) {
+      setSamples(
+        defaultSamples ||
+          getMaxSamplesForSpan(parsePrometheusDuration(state.span)),
+      );
     }
-  }, [defaultSamples, span]);
+  }, [defaultSamples, state.span]);
```

Comment:
This is a wrong use of a `useEffect` see https://react.dev/learn/you-might-not-need-an-effect

What you want is to set the samples whenever the default samples or the state.span is changing.

For that you need to do this: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes